### DISCUSSION TITLE
Was "Make useSearchParams and useParams compatible between app and pages router: #55280" ever added to a main release

### DISCUSSION THREAD

**Discussion by Judge-Paul** [vercel/next.js#57836] [Help]
### Summary

I am having an issue with useParams in pages router where I need to get the id from a dynamic route I set using
```
const params = useParams()
fetch(/api/params.id)
// Not actual API request just for illustration
```
and use this to make an API request using swr. The issue is that params.id is undefined and takes some seconds before being set and the API request will try to send params.id which will be undefined causing this to fail
I did some digging and found that a similar [issue](https://github.com/vercel/next.js/issues/54242) was already raised and a [PR](https://github.com/vercel/next.js/pull/55280) was made but this PR was only merged to a [canary release](https://github.com/vercel/next.js/releases/tag/v13.4.20-canary.37) and I don't think this was added to a main release cause I am still facing this same error.

**Comment by icyJoseph** [comment]
Not necessarily answering the main question, but you can do conditional fetching in SWR:

- https://swr.vercel.app/docs/conditional-fetching

So while params.id is undefined, you can making `skip` the request.

**Reply by Judge-Paul** [reply]
This won't have worked cause the id will still not be defined.
But I found out that if I use `useRouter` from `next/router` I can get the id
